hack/update-codegen.sh

```diff
@@ -31,8 +27,27 @@ else
   exit 0
 fi
 
+# This has been copied from https://github.com/bazelbuild/bazel/blob/master/tools/bash/runfiles/runfiles.bash
+# It exports rlocation function that can be used to access other bash scripts in this codebase.
+# set -e pipefail needs to happen after sourcing runfiles.bash, see https://github.com/bazelbuild/bazel/blob/master/tools/bash/runfiles/runfiles.bash#L21
+
+# --- begin runfiles.bash initialization v2 ---
+# Copy-pasted from the Bazel Bash runfiles library v2.
+set -uo pipefail; f=bazel_tools/tools/bash/runfiles/runfiles.bash
+source "${RUNFILES_DIR:-/dev/null}/$f" 2>/dev/null || \
+  source "$(grep -sm1 "^$f " "${RUNFILES_MANIFEST_FILE:-/dev/null}" | cut -f2- -d' ')" 2>/dev/null || \
+  source "$0.runfiles/$f" 2>/dev/null || \
+  source "$(grep -sm1 "^$f " "$0.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+  source "$(grep -sm1 "^$f " "$0.exe.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+  { echo>&2 "ERROR: cannot find $f"; exit 1; }; f=; set -e
+# --- end runfiles.bash initialization v2 ---
+
+
```

Comment:
This is so that we can source the functions that remove/replace the autogen build tags from another file (using the `rlocation` function that is being sourced here- I wanted to do that, to make this file shorter as I had to add some extra functionality to replacing/removing.
Arguably, this adds complexity, but is also an example of how bash libraries can be imported if that was ever useful.

This appears to be a standard way of doing it, see i.e https://github.com/pixie-io/pixie/blob/main/src/e2e_test/px_cluster/test_px_on_gke.sh#L33-L45

Reply:
It's pretty unreadable, but also quite interesting to look at.
It seems to be attempting to load `bazel_tools/tools/bash/runfiles/runfiles.bash` from various directories, which may be  derived from the name of the current script `$0`, right?

Thanks for the explanatory comment.

Reply:
Yes, it is interesting to look at!

> It seems to be attempting to load bazel_tools/tools/bash/runfiles/runfiles.bash from various directories, which may be derived from the name of the current script $0, right?

Yes, my understanding of this is that since bazel_tools/tools/bash/runfiles is declared as a dependency of the `update-codegen` Bazel target ([here](https://github.com/irbekrm/cert-manager/blob/bump_go_and_deps/hack/BUILD.bazel#L169)), it will get added to the 'runfiles' when we `bazel run` this target- see a limited description of that mechanism [here](https://docs.bazel.build/versions/main/be/shell.html#sh_binary.deps) and [here](https://docs.bazel.build/versions/main/test-encyclopedia.html#runfiles). These lines then attempt to source the [`runfiles.bash`](https://github.com/bazelbuild/bazel/blob/master/tools/bash/runfiles/runfiles.bash) script from one of the known locations of these Bazel runfiles.

Perhaps I should add a comment about this.